File: source/bxgenbb_help/src/decay0/Sn120low.cc

#include <sstream>
#include <stdexcept>
#include <cmath>

#include <mygsl/rng.h>

#include <genbb_help/decay0/Sn120low.h>
#include <genbb_help/primary_event.h>
#include <genbb_help/decay0/alpha.h>
#include <genbb_help/decay0/gamma.h>
#include <genbb_help/decay0/electron.h>
#include <genbb_help/decay0/positron.h>
#include <genbb_help/decay0/particle.h>
#include <genbb_help/decay0/pair.h>
#include <genbb_help/decay0/nucltransK.h>
#include <genbb_help/decay0/nucltransKL.h>
#include <genbb_help/decay0/nucltransKLM.h>
#include <genbb_help/decay0/nucltransKLM_Pb.h>
#include <genbb_help/decay0/beta.h>
#include <genbb_help/decay0/beta1.h>
#include <genbb_help/decay0/beta2.h>
#include <genbb_help/decay0/beta_1fu.h>
#include <genbb_help/decay0/PbAtShell.h>

namespace genbb {
  namespace decay0 {

    // Sn120low.f 
    // This file was extracted from the 'decay0' program by V.I. Tretyak
    // Copyright 1995-2011 V.I. Tretyak
    // This program is free software
    // it under the terms of the GNU General Public License as published by
    // the Free Software Foundation
    // your option) any later version.
    // 
    // This program is distributed in the hope that it will be useful, but
    // WITHOUT ANY WARRANTY
    // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
    // General Public License for more details.
    // 
    // You should have received a copy of the GNU General Public License
    // along with this program
    // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
    // 

    void Sn120low(mygsl::rng & prng_, genbb::primary_event & event_, int levelkev)
    {
      //double t;
      double tdlev;
      double tclev;
      double thlev;
      // Subroutine describes the deexcitation process in Sn120 nucleus
      // after 2b-decay of Te120 to ground 0+ and excited 2+ levels
      // of Sn120 ("Table of Isotopes", 7th ed., 1978).
      // Call : call Sn120low(levelkev)
      // Input : levelkev - energy of Sn120 level (integer in keV) occupied
      // initially
      // 0+(gs) - 0 keV,
      // 2+(1) - 1171 keV.
      // Output: // common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),// ptime(100).
      // VIT, 13.02.2004.
      tclev=0.;
      if (levelkev == 1171) goto label_1171;
      if (levelkev ==  0) goto label_10000;
      goto label_20000;
    label_1171  :
      thlev=0.642e-12;
      decay0_nucltransK(prng_, event_, 1.171,0.029,8.0e-4,0.,tclev,thlev,tdlev);
      return;
    label_10000  :
      return;
    label_20000  :
      // print *,'Sn120: wrong level [keV] ',levelkev
      return;
    }
    // end of Sn120low.f




  } // end of namespace decay0 
} // end of namespace genbb 

// end of Sn120low.cc
// Local Variables: --
// mode: c++ --
// End: --

File: source/bxgenbb_help/src/decay0_fortran/Sn120low.f
c Sn120low.f 
c
c This file was extracted from the 'decay0' program by V.I. Tretyak
c
c Copyright 1995-2011 V.I. Tretyak
c
c This program is free software; you can redistribute it and/or modify
c it under the terms of the GNU General Public License as published by
c the Free Software Foundation; either version 3 of the License, or (at
c your option) any later version.
c 
c This program is distributed in the hope that it will be useful, but
c WITHOUT ANY WARRANTY; without even the implied warranty of
c MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
c General Public License for more details.
c 
c You should have received a copy of the GNU General Public License
c along with this program; if not, write to the Free Software
c Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
c 
c
	subroutine Sn120low(levelkeV)
c Subroutine describes the deexcitation process in Sn120 nucleus
c after 2b-decay of Te120 to ground 0+ and excited 2+ levels
c of Sn120 ("Table of Isotopes", 7th ed., 1978).
c Call  : call Sn120low(levelkeV)
c Input : levelkeV - energy of Sn120 level (integer in keV) occupied
c                    initially; following levels can be occupied:
c                    0+(gs) -    0 keV,
c                    2+(1)  - 1171 keV.
c Output: common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),ptime(100).
c VIT, 13.02.2004.
	tclev=0.
	if(levelkev.eq.1171) go to  1171
	if(levelkev.eq.   0) go to 10000
	                     go to 20000
c-----------------------------------------------------------------------
1171	thlev=0.642e-12
	call nucltransK(1.171,0.029,8.0e-4,0.,tclev,thlev,tdlev)
	return
c-----------------------------------------------------------------------
10000	return
c-----------------------------------------------------------------------
20000	print *,'Sn120: wrong level [keV] ',levelkev
c-----------------------------------------------------------------------
	return
	end
c
c end of Sn120low.f
c
c Local Variables: --
c mode: fortran --
c End: --
